### ISSUE TITLE
NeoVim autocompletion problem

### ISSUE THREAD

**Issue by ram535** [natebosch/dart_language_server#8]
I tried to configure dart_language_server with NeoVim but I got some issues.
So far dart_language_server marks erros and warnings on the file but I don't get autocompletion.

This is the configuration I use in NeoVim.

```
call plug#begin('~/.config/nvim/plugged')
  Plug 'autozimu/LanguageClient-neovim', { 'do': ':UpdateRemotePlugins' }
  Plug 'roxma/nvim-completion-manager'
  call plug#end()

let g:LanguageClient_serverCommands = {
    \ 'dart': ['dart_language_server'],
    \ }
```

**Comment by natebosch** [comment]
I haven't used neovim or the LanugageClient plugin. A cursory search isn't turning up any extra config required... but the documentation is stale - the readme mentions `nvim-completion-manager` but then the docs say "When using a supported completion manager (currently only deoplete supported), completion should work out of the box."

A few things to check on:

- Does completion work with other language servers?
- Do other completion-manager sources work?
- If you have a file with no sensitive content you could attach a log file and we can look for the completion request. It would be the one with `"method":"textDocument/completion"`. The log has a name like `/tmp/lsp-wirelog-<something>.log`

**Comment by autozimu** [comment]
Please also refer https://github.com/autozimu/LanguageClient-neovim/issues/40

The issue ram535 encountered was that dart language server is expecting `trace` property exists in initialize params, while by spec, the property is optional and default to `off`.

I've just updated LanguageClient-neovim to provide default trace setting, but I believe in general it would be good to conform to spec.

(by the way, thanks for pointing out the outdated doc.)

**Comment by ram535** [comment]
Thanks. I tested your changes and now autocompletion is working.

**Comment by natebosch** [comment]
Thanks for the pointer on `trace`! I've added some defaults.

Glad to hear you've got it working, I've also published version 0.0.6 with the server side fix.
